Source/WebCore/editing/DeleteSelectionCommand.cpp

```diff
@@ -423,10 +429,15 @@ void DeleteSelectionCommand::saveTypingStyleState()
     RefPtr<Node> startNode = m_selectionToDelete.start().deprecatedNode();
     if (!startNode->isTextNode() && !startNode->hasTagName(imgTag) && !startNode->hasTagName(brTag))
         return;
+    
+    if (shouldNotInheritStyleFrom(*m_selectionToDelete.start().deprecatedNode()))
+        return;
 
     // Figure out the typing style in effect before the delete is done.
     m_typingStyle = EditingStyle::create(m_selectionToDelete.start(), EditingStyle::EditingPropertiesInEffect);
-    m_typingStyle->removeStyleAddedByNode(enclosingAnchorElement(m_selectionToDelete.start()));
+    m_typingStyle->removeStyleAddedByNode(enclosingAtomicNode(m_selectionToDelete.start()));
+
+    return enclosingNodeOfType(m_selectionToDelete.start(), isAtomicNode);
```

Comment:
This shouldn't be here. It should be in the definition of enclosingAtomicNode.

Reply:
Added it in `Editing.cpp` with return for this similar to `enclosingAnchorElement`.

Reply:
Compiling and running through EWS. Let's see.